### PR TITLE
feat: Add a config for openedx-events annotations

### PR DESCRIPTION
**Pull request by bmtcril**
**Description:** Adding code and config to find and document annotations used in the openedx-events repository. When run in openedx-events (see https://github.com/openedx/openedx-events/pull/318) the generated documentation looks like this:

Screenshot 2024-02-20 at 1 11 21 PM

### CODE REVIEW

**Review by timmc-edx — APPROVED** [review]
My review shouldn't be considered authoritative as I'm not a maintainer and I'm not familiar with this repo, but -- I did review the fields/annotations for completeness and consistency.

File: code_annotations/contrib/sphinx/extensions/openedx_events.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            event_name_literal = nodes.literal(text=event_name)
+            event_data = event[".. event_data:"]
+            event_data_literal = nodes.literal(text=event_data)
+            event_key_field = event.get(".. event_key_field:", "")
+            event_key_literal = nodes.literal(text=event_key_field)
+            event_description = event[".. event_description:"]
+
+            event_section = nodes.section("", ids=[f"openedxevent-{event_type}"])
+            event_section += nodes.title(text=event_type, ids=[f"title-{event_type}"])
+            event_section += nodes.paragraph(text=f"Description:"
```
Comment (mariajgrimaldi) [inline code_annotations/contrib/sphinx/extensions/openedx_events.py:106]: Maybe:
```suggestion
            event_section += nodes.paragraph(text=f"Description: "
```

File: code_annotations/contrib/sphinx/extensions/openedx_events.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                                                  f"{event_description}")
+            event_section += nodes.paragraph(" ", "Signal name:", event_name_literal)
+            if event_key_field:
+                event_section += nodes.paragraph(
+                    " ",
+                    "Event key field: ",
+                    event_key_literal
+                )
+            event_section += nodes.paragraph(" ", "Event data:", event_data_literal)
```
Comment (mariajgrimaldi) [inline code_annotations/contrib/sphinx/extensions/openedx_events.py:115]:
```suggestion
            event_section += nodes.paragraph(" ", "Event data: ", event_data_literal)
```

**Comment by bmtcril** [comment]
@mariajgrimaldi thanks, the docs said that first argument was supposed to be a separator so I thought it would... do that. :) All set now.

File: code_annotations/contrib/sphinx/extensions/openedx_events.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+from .base import find_annotations
+
+
+def find_events(source_path):
+    """
+    Find the feature toggles as defined in the configuration file.
+
+    Return:
+        toggles (dict): feature toggles indexed by name.
+    """
```
Comment (mariajgrimaldi) [inline code_annotations/contrib/sphinx/extensions/openedx_events.py:20]: Shoud we update this docstring?

Comment (bmtcril) [reply code_annotations/contrib/sphinx/extensions/openedx_events.py]: Oof, yes. Done.

**Review by mariajgrimaldi — APPROVED** [review]
@timmc-edx: this implementation is similar to the featuretoggles one https://github.com/openedx/code-annotations/blob/786da8c202eec47f8775c7691718b75fbe32550d/code_annotations/contrib/sphinx/extensions/featuretoggles.py#L26 with implementation-specific changes to generate events docs instead. I understand you're not the maintainer of the repo, so could we include someone familiar with the code? Because of the similarity with the other extensions, I don't think it'd be difficult to review.

**Comment by timmc-edx** [comment]
I've attempted to ping the owning team within 2U, but haven't heard back yet. I think if another week goes by without hearing from them, we should just merge this.

**Comment by timmc-edx** [comment]
@bmtcril I think we're good to merge this.

**Pull request merged**